### PR TITLE
chore: Remove dependency on `transforms-json_parser`

### PR DESCRIPTION
**Pull request by ktff**
Ref. #1940 

PR doesn't extract shared code, but specializes it for it's call site. It's a bit faster, shorter, clearer, and has a dependency less.

### CODE REVIEW

File: src/sources/kubernetes_logs/parser/docker.rs [hunk 1]
```diff
@@ -22,32 +20,29 @@ lazy_static! {
...
+    let to_parse = log.remove(&event::log_schema().message_key())?.as_bytes();
+
+    match serde_json::from_slice(to_parse.as_ref()) {
+        Ok(JsonValue::Object(object)) => {
+            for (key, value) in object {
+                log.insert_flat(key, value);
+            }
+            Some(())
+        }
+        Ok(_) | Err(_) => None,
```
Comment (MOZGIII) [inline src/sources/kubernetes_logs/parser/docker.rs:45]: We probably want to emit an event here with a counter.

Comment (ktff) [reply src/sources/kubernetes_logs/parser/docker.rs]: Like this?

File: src/sources/kubernetes_logs/parser/docker.rs [hunk 2]
```diff
@@ -42,7 +43,10 @@ fn parse_json(log: &mut LogEvent) -> Option<()> {
             }
             Some(())
         }
-        Ok(_) | Err(_) => None,
+        Ok(_) | Err(_) => {
+            emit!(KubernetesLogsMessageParseFailed { message: &to_parse });
```
Comment (MOZGIII) [inline src/sources/kubernetes_logs/parser/docker.rs:47]: Looks good, but let's put the `Docker` somewhere in the name. This should be a docker-specific parser error - pretty much mirroring the similar emit at JSON.

How about `KubernetesLogsDockerFormatParseFailed`?

Similarly, the `counter!` can emit `k8s_docker_format_parse_failures`.

Comment (ktff) [reply src/sources/kubernetes_logs/parser/docker.rs:47]: Sounds good.

**Pull request merged**